open_alchemy/build/__init__.py

```diff
@@ -218,16 +250,105 @@ def dump(
         raise exceptions.BuildError(str(exc)) from exc
 
 
-def execute(*, spec: typing.Any, name: str, path: str) -> None:
+def build_dist(name: str, path: str, format_: PackageFormat) -> None:
+    """
+    Build a distibution package.
+
+    The formats can be combined with the bitwise operator "or" (``|``), for
+    instance, building both sdist and wheel packages can be specified like this:
+
+    .. code-block:: python
+
+        format_=PackageFormat.SDIST|PackageFormat.WHEEL
+
+    Args:
+        path: The package directory.
+        format: The package format to build.
+    """
+    pkg_dir = pathlib.Path(path) / name
+    if PackageFormat.SDIST in format_:
+        build_sdist(str(pkg_dir))
+    if PackageFormat.WHEEL in format_:
+        build_wheel(str(pkg_dir))
+
+
+def build_sdist(path: str) -> None:
+    """
+    Build a .tar.gz source distribution and place it in a "dist" folder.
+
+    The "dist" folder is located at the root of the project.
+
+    Args:
+        path: The package directory.
+    """
+    run(f"{sys.executable} setup.py sdist", path)
+
+
+def build_wheel(path: str) -> None:
+    """
+    Build a .whl package and place it in a "dist" folder.
+
+    The "dist" folder is located at the root of the project.
+
+    Args:
+        path: The package directory.
+    """
+    try:
+        run(f"{sys.executable} setup.py bdist_wheel", path)
+    except exceptions.BuildError as exc:
+        raise RuntimeError(
+            "Building a wheel package requires the wheel package. "
+            "Try `pip install wheel`."
+        ) from exc
+
+
+def run(cmd: str, cwd: str) -> typing.Tuple[str, str]:
+    """
+    Run a shell command.
+
+    Args:
+        cmd: The command to execute.
+        cwd: The path where the command must be executed from.
+
+    Returns:
+        A tuple containing (stdout, stderr).
+    """
+    try:
+        # "nosec" is used here as we believe we followed the guidelines to use
+        # subprocess securely:
+        # https://security.openstack.org/guidelines/dg_use-subprocess-securely.html
+        output = subprocess.run(  # nosec
+            shlex.split(cmd),
+            cwd=cwd,
+            check=True,
+            shell=False,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+    except subprocess.CalledProcessError as exc:
+        raise exceptions.BuildError(str(exc)) from exc
+    else:
+        return output.stdout.decode("utf-8"), output.stderr.decode("utf-8")
+
+
+def execute(
+    *,
+    spec: typing.Any,
+    name: str,
+    path: str,
+    format_: PackageFormat = PackageFormat.NONE,
```

Comment:
Did you make a decision on wanting to keep the default value in place?

Reply:
Ah yes, I forgot this one. So since our `build_*()` functions already have a default value for the format, I took it away from the `execute()` function.